app/domain/operations/eligible/build_grant.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require "dry/monads"
+require "dry/monads/do"
+
+module Operations
+  module Eligible
+    # Operation to support eligibility creation
+    class BuildGrant
+      include Dry::Monads[:do, :result]
+
+      # @param [Hash] opts Options to build eligibility
+      # @option opts [<Symbol>]   :grant_key required
+      # @option opts [<String>]   :grant_value required
+      # @return [Dry::Monad] result
```

Comment:
It's not clear to me from comments what this returns, beyond a monadic Success/Failure. Perhaps we can indicate what the happy-path Success contents are?